polynote-frontend/polynote/plot_editor.js

```diff
@@ -223,28 +303,35 @@ export class PlotEditor extends EventTarget {
             );
         }
 
-        if(ops.length) {
-            return this.session.request(new ModifyStream(this.path, this.repr.handle, ops, null)).then(mod => mod.newRepr)
-        }
-
-        return Promise.as(this.repr)
+        return ops;
     }
 
-    addXDimension(from) {
-        // TODO: multiple dimensions? Custom groupings?
-        const field = from.field;
+    set xField(field) {
         this.xDimension = field;
+        this.xAxisDrop.classList.add('nonempty');
         const label = this.xAxisDrop.querySelector('.label');
-        label.innerHTML = '';
-        label.appendChild(document.createTextNode(field.name));
+        [...label.querySelectorAll('.numeric, .dimension')].forEach(node => node.parentNode.removeChild(node));
+        label.appendChild(span([this.correctXType], [field.name]));
+    }
+
+    addXField(from) {
```

Comment:
What's the idea behind both `addXField` and `set xField`? (also, is `this.xField` used anywhere?)

Reply:
Yeh, kind of a mess... I removed the setter thing and just changed it to `setXField`, and removed `addXField`.